sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -222,7 +225,7 @@ case class PreprocessTableCreation(sparkSession: SparkSession) extends Rule[Logi
     val columnNames = if (sparkSession.sessionState.conf.caseSensitiveAnalysis) {
       schema.map(_.name)
     } else {
-      schema.map(_.name.toLowerCase)
+      schema.map(_.name.toLowerCase(Locale.ROOT))
```

Comment:
Maybe, it is not good to point the similar instances all but let me just point this out as the change looks big. This maybe the similar instances with https://github.com/apache/spark/pull/17527/files#r110317272.